weblate/accounts/pipeline.py

```diff
@@ -379,17 +383,18 @@ def store_email(strategy, backend, user, social, details, **kwargs):
     # The email can be empty for some services
     if details.get("verified_emails"):
         current = set(details["verified_emails"])
-        existing = set(social.verifiedemail_set.values_list("email", flat=True))
+        existing = set(social.verifiedemail_set.values_list("email", "is_deliverable"))
         for remove in existing - current:
-            social.verifiedemail_set.filter(email=remove).delete()
+            social.verifiedemail_set.filter(email=remove[0]).delete()
```

Comment:
The filter should use both params (to avoid issue when e-mail deliverability changes):

```suggestion
            social.verifiedemail_set.filter(email=remove[0], is_deliverable=remove[1]).delete()
```